### ISSUE TITLE
rootDir does not get resolved/interpolated correctly

### ISSUE THREAD

**Issue by Ionaru** [Romakita/ts-express-decorators#94]
## Informations

Type | Version
---|---
Bug | 2.0.12

****

## Description
The rootDir does not get resolved correctly.

I followed the guide on https://romakita.github.io/ts-express-decorators/#/getting-started and besides some small tweaks to make it actually work, I found that the rootDir does not get resolved and placed correctly in the scanner paths.

```
"C:\Program Files\JetBrains\IntelliJ IDEA 2017.2\bin\runnerw.exe" "C:\Program Files\nodejs\node.exe" "C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js" run start --scripts-prepend-node-path=auto

> test-ts-express-decorator@1.0.0 start C:\Users\Jeroen\IdeaProjects\express-typescript
> concurrently "npm run tsc:w" "nodemon server.js --ignore *.ts"

[1] [nodemon] 1.11.0
[1] [nodemon] to restart at any time, enter `rs`
[1] [nodemon] watching: *.*
[1] [nodemon] starting `node server.js`
[0] 
[0] > test-ts-express-decorator@1.0.0 tsc:w C:\Users\Jeroen\IdeaProjects\express-typescript
[0] > tsc -w
[0] 
[1] [2017-08-28 21:15:01.622] [INFO ] [TSED] - Scan files : C:\Users\Jeroen\IdeaProjects\express-typescript\${rootDir}\controllers\**\*.js
[1] [2017-08-28 21:15:01.627] [INFO ] [TSED] - Scan files : C:\Users\Jeroen\IdeaProjects\express-typescript\middlewares\**\*.js
[1] [2017-08-28 21:15:01.628] [INFO ] [TSED] - Scan files : C:\Users\Jeroen\IdeaProjects\express-typescript\services\**\*.js
[1] [2017-08-28 21:15:01.628] [INFO ] [TSED] - Scan files : C:\Users\Jeroen\IdeaProjects\express-typescript\converters\**\*.js
[1] [2017-08-28 21:15:01.632] [INFO ] [TSED] - settings.rootDir => C:\Users\Jeroen\IdeaProjects\express-typescript
[1] [2017-08-28 21:15:01.632] [INFO ] [TSED] - settings.env => development
[1] [2017-08-28 21:15:01.632] [INFO ] [TSED] - settings.httpPort => 8080
[1] [2017-08-28 21:15:01.632] [INFO ] [TSED] - settings.httpsPort => 8000
[1] [2017-08-28 21:15:01.632] [INFO ] [TSED] - settings.endpointUrl => /rest
[1] [2017-08-28 21:15:01.632] [INFO ] [TSED] - settings.version => 1.0.0
[1] [2017-08-28 21:15:01.632] [INFO ] [TSED] - settings.uploadDir => ${rootDir}/uploads
[1] [2017-08-28 21:15:01.633] [INFO ] [TSED] - settings.debug => false
[1] [2017-08-28 21:15:01.633] [INFO ] [TSED] - settings.authentification => undefined
[1] [2017-08-28 21:15:01.633] [INFO ] [TSED] - settings.mount => { '/': '${rootDir}/controllers/**/*.js' }
[1] [2017-08-28 21:15:01.635] [INFO ] [TSED] - settings.componentsScan => [ '${rootDir}/middlewares/**/*.js',
[1]   '${rootDir}/services/**/*.js',
[1]   '${rootDir}/converters/**/*.js' ]
[1] [2017-08-28 21:15:01.636] [INFO ] [TSED] - settings.acceptMimes => [ 'application/json' ]
[1] [2017-08-28 21:15:01.636] [INFO ] [TSED] - Build services
[1] [2017-08-28 21:15:01.707] [INFO ] [TSED] - Build controllers
[1] [2017-08-28 21:15:01.708] [INFO ] [TSED] - Routes mounted :
[1] [2017-08-28 21:15:01.708] [INFO ] [TSED] - 
[1] ┌────────┬──────────┬──────────────┐
[1] │ Method │ Endpoint │ Class method │
[1] └────────┴──────────┴──────────────┘
[1] [2017-08-28 21:15:01.730] [INFO ] [TSED] - HTTP Server listen on http://0.0.0.0:8080
[1] [2017-08-28 21:15:01.731] [INFO ] [TSED] - HTTP Server listen on https://0.0.0.0:8000
[1] Server started...
[1] [2017-08-28 21:15:01.733] [INFO ] [TSED] - Started in 103 ms
```

As you can see, at one point it shows the controller path as `C:\Users\Jeroen\IdeaProjects\express-typescript\${rootDir}\controllers\**\*.js` and another time as `'${rootDir}/controllers/**/*.js'`.

The odd thing is that the other directories do get resolved correctly, and if I place my `calendarCtrl.ts` in there and adjust the ServerSettings, it works.

**Comment by Ionaru** [comment]
I was able to get it running by changing my ServerSettings to
```
@ServerSettings({
    rootDir: Path.resolve(__dirname),
    mount: {
        "/": "controllers/**/*.js"
    },
    componentsScan: [
        "middlewares/**/*.js",
        "services/**/*.js",
        "converters/**/*.js"
    ],
    acceptMimes: ["application/json"]
})
```

**Comment by Romakita** [comment]
Hi @Ionaru ,

Wich version of Windows have you ?

See you, 
Romain

**Comment by Ionaru** [comment]
I'm using Windows 10, version 1703, build 16251.1002.
But I've had the same issue on my Linux Kubuntu 17.10 machine.

**Comment by Romakita** [comment]
Thanks @Ionaru, I'll try to reproduced your bug and fixed it :)

Ok. I haven't issue on mac or travis (linux). But I thinks it's a real bug ;)
Can you send me the exact setting server configuration that produce the bug ?

**Comment by Ionaru** [comment]
```
@ServerSettings({
    rootDir: Path.resolve(__dirname),
    acceptMimes: ["application/json"]
})
```

Those settings will reproduce the bug, it's an exact copy from the 'getting-started' guide at https://romakita.github.io/ts-express-decorators/#/getting-started

**Comment by Romakita** [comment]
It's bug. I've the same issue with the single quote. I'll fixed it ASAP ;)

**Comment by Ionaru** [comment]
Awesome, thanks!

**Comment by Romakita** [comment]
@Ionaru 
Fixed with the v2.0.13 :)
